### PR TITLE
WIP - feat(Core/Map): 3.3.5 Mythic Plus

### PR DESCRIPTION
**Pull request by P-Kito**
grafik

Core/Game: Add possibility to add custom Map Difficulties, implement Blizzards unused Difficulty EPIC and reuse it for Mythic+ 335

## CHANGES PROPOSED:
-  
-  

## ISSUES ADDRESSED:
- Closes 

## TESTS PERFORMED:
-
-

## KNOWN ISSUES AND TODO LIST:

- [ ]
- [ ] 

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by BarbzYHOOL** [comment]
you could make it a pr draft too

File: data/sql/updates/pending_db_world/rev_1588196804255396600.sql [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+DROP TABLE IF EXISTS `map_difficulty_data`;
+CREATE TABLE IF NOT EXISTS `map_difficulty_data` (
+  `MapId` INT(10) UNSIGNED NOT NULL DEFAULT 0,
+  `Difficulty` INT(10) UNSIGNED NOT NULL DEFAULT 0,
+  `ParentDifficulty` INT(10) UNSIGNED NOT NULL DEFAULT 0,
+  `Comment` TEXT,
+  PRIMARY KEY (`MapId`, `Difficulty`)
+) ENGINE=MYISAM;
+
```
Comment (BarbzYHOOL) [inline data/sql/updates/pending_db_world/rev_1588196804255396600.sql:11]: is this table only used to set epic difficulty? if so then the name should reflect that else it's confusing

Comment (BarbzYHOOL) [reply data/sql/updates/pending_db_world/rev_1588196804255396600.sql:11]: also the columns naming convention should follow the table naming convention aka lowercase with _

Comment (P-Kito) [reply data/sql/updates/pending_db_world/rev_1588196804255396600.sql:11]: No, it does exactly what it says ;-)
It extends the map_difficulty_data by the entries

**Comment by kuramajssken001** [comment]
This looks cool, but we should be more in line with blizzard's original design.

**Comment by P-Kito** [comment]
Yop, im closing this and gonna develop it privately.

File: data/sql/updates/pending_db_world/rev_1588196804255396600.sql (hunk 1 above)
Comment (BarbzYHOOL) [reply data/sql/updates/pending_db_world/rev_1588196804255396600.sql:11]: but there is only one difficulty

it is just straight confusing

people might think it's related to heroic or raid size dunno

**Comment by FrancescoBorzi** [comment]
what about making this into a module? I don't think it's a good idea to push custom things in the core :)

**Comment by Yehonal** [comment]
As I said into the PM you can develop it with a module, you explained to me that you need the special EPIC DIFFICULTY to make it work. I think it can be added into the core as an optional config to enable since it's something that can unlock hidden client features (I agree with it)

But the mythic+ system itself isn't something that must be added to the core, it's neither blizzlike nor an hidden/configurable game feature so must stay out of the core.

As you can see from the following pictures, I've implemented the mythic+ with a module based on autobalance one (using its internal hooks). So it's totally doable. I think I'll release such module in future but would be nice to merge both ideas

image

image

**Pull request closed without merging**